interfaces/connection.go

```diff
@@ -45,6 +46,39 @@ type ConnectedSlot struct {
 	dynamicAttrs map[string]interface{}
 }
 
+// AttrGetter is an interface with Attr getter method common
+// to ConnectedSlot, ConnectedPlug, PlugInfo and SlotInfo types.
+type AttrGetter interface {
+	Attr(key string) (interface{}, error)
+}
+
+func getAttribute(staticAttrs map[string]interface{}, dynamicAttrs map[string]interface{}, key string, val interface{}) error {
+	var v interface{}
+	var ok bool
+
+	if dynamicAttrs != nil {
```

Comment:
The three lines above can go away and be replaced with just:
```
v, ok := dynamicAttrs[key]
```
This will work as you want if the map is nil.